src/Components/Overlay/Overlay.tsx

```diff
@@ -46,51 +53,71 @@ export const OverlayRoot: FunctionComponent<OverlayProps> = ({
     [`${className}`]: className,
   })
 
-  const overlay = (
-    <>
-      <Transition
-        items={visible}
-        from={{opacity: 0}}
-        enter={{opacity: 0.7}}
-        leave={{opacity: 0}}
-        config={transitionConfig}
-      >
-        {visible => visible && (props => renderMaskElement(props))}
-      </Transition>
-      <Transition
-        items={visible}
-        from={{opacity: 0, transform: 'translateY(44px)'}}
-        enter={{opacity: 1, transform: 'translateY(0)'}}
-        leave={{opacity: 0, transform: 'translateY(44px)'}}
-        config={transitionConfig}
-      >
-        {visible =>
-          visible &&
-          (props => (
-            <DapperScrollbars
-              className={overlayClass}
-              thumbStartColor={InfluxColors.White}
-              thumbStopColor={InfluxColors.Hydrogen}
-              noScrollX={true}
-              autoHide={false}
-              testID={testID}
-              id={id}
-            >
-              <div
-                className="cf-overlay--children"
-                data-testid={`${testID}--children`}
-                style={props}
+  const handleEscapeKey = (e: KeyboardEvent): void => {
+    if (e.key === 'Escape' && visible && onEscape) {
+      onEscape(false)
+    }
+  }
+
+  if (!visible) {
+    return null
+  }
```

Comment:
Is it possible to make this change work without returning `null`? It sidesteps all the work that went into animating the overlays in/out